### PR TITLE
Allow a custom errorHandler in parallel + tryCatch

### PR DESCRIPTION
**Pull request by lukearndt**
We want to be able to supply a custom errorHandler so that we can link to error
reporting services.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -14,17 +14,21 @@ import {
   take
 } from "redux-saga/effects";
 
+const defaultErrorHandler = function(error) {
+  console.error("Error caught by redux-saga-catch: ", error);
```
Comment (afestein) [inline src/index.js]: I don't think we want any calls to `console` inside our app as it significantly degrades performance.

https://facebook.github.io/react-native/docs/performance#using-consolelog-statements

File: src/index.js [hunk 2]
```diff
@@ -14,17 +14,21 @@ import {
...
  * @param {Saga} saga
  */
-export function tryCatch(saga) {
+export function tryCatch(saga, errorHandler = defaultErrorHandler) {
   const wrapped = function* wrappedTryCatch() {
     try {
       yield call(saga, ...arguments);
     } catch (e) {
-      console.error("Error caught by redux-saga-catch: ", e);
+      errorHandler(e);
```
Comment (afestein) [inline src/index.js]: Doesn't Prettier remove the semi-colons?

Comment (lukearndt) [reply src/index.js]: This library doesn't adhere to the same style rules as yesheis.

File: src/index.js (hunk 1 above)
Comment (lukearndt) [reply src/index.js]: I expect we will supply a custom error handler for yesheis.

File: src/index.js [hunk 3]
```diff
@@ -14,17 +14,21 @@ import {
...
+  console.error("Error caught by redux-saga-catch: ", error);
+};
+
 /**
  * Quick wrap a saga with `try catch`
  * 快速使用try catch包装saga
  * @param {Saga} saga
  */
-export function tryCatch(saga) {
+export function tryCatch(saga, errorHandler = defaultErrorHandler) {
```
Comment (parndt) [inline src/index.js]: should this be a named argument a la 
```js
export function tryCatch(saga, { errorHandler: defaultErrorHandler })
```
?

File: src/index.js (hunk 1 above)
Comment (afestein) [reply src/index.js]: Now that I think about it, I don't think that performance is really an issue - if it gets to here, then the app has probably already crashed 😄

File: src/index.js [hunk 4]
```diff
@@ -73,11 +78,12 @@ export function throttle(ms, pattern, worker, ...args) {
  * usage/用法:
  * yield parallel([function*(){}, ...sagas])
  * @param {Saga[]} sagas
+ * @param {Function} errorHandler
  */
-export function parallel(sagas) {
+export function parallel(sagas, errorHandler) {
```
Comment (parndt) [inline src/index.js]: errorHandler should be optional to maintain the API

**Pull request merged**